app/src/main/java/com/amaze/filemanager/activities/ImageViewerActivity.java

```diff
@@ -0,0 +1,298 @@
+        }
+    }
+
+    private void loadImageFileList() {
+        baseFiles = new ArrayList<>();
+        final int IS_FOLDER = 1;
+
+        if (FileUtil.checkFolder(currentFile.getParent(this), this) == IS_FOLDER) {
+            BaseFile parentDir = new BaseFile(currentFile.getParent(this));
+            int index = 0;
+            for (BaseFile baseFile : parentDir.listFiles(this, parentDir.isRoot())) {
+                if (baseFile.getMimeType() != null) {
+                    if (baseFile.getPath().equals(currentFile.getPath())) {
+                        currentIndex = index;
+                    }
+                    if (baseFile.getMimeType().contains("image/")) {
+                        baseFiles.add(baseFile);
+                        index++;
+                    }
+                }
+            }
+        }
+    }
+
+    private void initToolbar() {
+        toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+    }
+
+    private void initIvImageView() {
+        ivImage = (TouchImageView) findViewById(R.id.ivImage);
+        loadImage();
+
+        ivImage.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                if (ivImage.getCurrentZoom() > NO_ZOOM) {
+                    return false;
+                }
+
+                switch (event.getAction()) {
+                    case MotionEvent.ACTION_DOWN:
+                        oldXValue = event.getX();
+                        break;
+
+                    case MotionEvent.ACTION_UP:
+                        if (ivImage.getCurrentZoom() != NO_ZOOM) {
+                            return false;
+                        }
+                        float currentX = event.getX();
+
+                        if (currentX > oldXValue) {
+                            //SWIPE RIGHT
+                            showPreviousImage();
+                        }
+
+                        if (currentX < oldXValue) {
+                            //SWIPE LEFT
+                            showNextImage();
+                        }
+
+                        break;
+                }
+                return true;
+            }
+        });
+
+        ivImage.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (llToolbarLayout.getVisibility() == View.VISIBLE) {
+                    setMenuVisibility(View.INVISIBLE);
+                } else {
+                    setMenuVisibility(View.VISIBLE);
+                }
+            }
+        });
+    }
+
+    private void setMenuVisibility(int visible) {
+        llToolbarLayout.setVisibility(visible);
+        llBottomBar.setVisibility(visible);
+    }
+
+    private void loadImage() {
+        GlideApp.with(ivImage)
+                .load(currentFile.getPath())
+                .error(R.mipmap.ic_launcher)
```

Comment:
Does this mean that ic_launcher will be shown if Glide fails to load the correct one? You should snackbar an error like "Error loading image" and show a grey/white cross.